### ISSUE TITLE
Failed to analyze project: failed to find the main target of the scheme

### ISSUE THREAD

**Issue by rachid-debu** [bitrise-steplib/steps-ios-auto-provision-appstoreconnect#7]
Hello

I'm build a cordova based iOS app.
I'm using the `Scripts` step to do the build (because we have some commands to be ran). Build succeeds.

My problem is afterwards, with `ios-auto-provision-appstoreconnect` i'm getting the error below : 

`Failed to analyze project: failed to find the main target of the scheme (Doyouno): failed to find scheme (Doyouno) in project: scheme Doyouno not found in Doyouno`

Here's a link to my build : https://app.bitrise.io/build/1717dca025d9baad#?tab=log 

Would highly appreciate some help.

Let me know if need more informations.

**Comment by bitce** [comment]
Thanks, @rachid-debu! Can you provide some context on the setup you do? Is the .xcworkspace file committed to the repository?

**Comment by rachid-debu** [comment]
@bitce For info, I've downgraded to the prevision version of `steps-ios-auto-provision-appstoreconnect` --> `steps-ios-auto-provision` and provided the obsolete authentication method. My build is working fine now.

Regarding this new version, I didn't manage to get this working. I'll stick to the older one, until further notice.

**Comment by mac89** [comment]
I'm experiencing the same issue with a Cordova generated project. The file structure:
```
+ file='/Users/vagrant/git/platforms/ios/MTinfo 3000 TST.xcworkspace'
+ ls -R -ltr '/Users/vagrant/git/platforms/ios/MTinfo 3000 TST.xcworkspace'
total 8
drwxr-xr-x  3 vagrant  staff   96 Jun 25 14:19 xcshareddata
-rw-r--r--  1 vagrant  staff  233 Jun 25 14:21 contents.xcworkspacedata
/Users/vagrant/git/platforms/ios/MTinfo 3000 TST.xcworkspace/xcshareddata:
total 0
drwxr-xr-x  3 vagrant  staff  96 Jun 25 14:19 xcschemes
/Users/vagrant/git/platforms/ios/MTinfo 3000 TST.xcworkspace/xcshareddata/xcschemes:
total 8
-rw-r--r--  1 vagrant  staff  3399 Jun 25 14:19 MTinfo 3000 TST.xcscheme
```

This step's output:

```
Config:
- BuildAPIToken: [REDACTED]
- BuildURL: https://app.bitrise.io/build/b5ce7f49e9e3eb70
- ProjectPath: /Users/vagrant/git/platforms/ios/MTinfo 3000 TST.xcworkspace
- Scheme: MTinfo 3000 TST
- Configuration: 
- Distribution: app-store
- MinProfileDaysValid: 0
- CertificateURLList: [REDACTED]
- CertificatePassphraseList: 
- KeychainPath: /Users/vagrant/Library/Keychains/login.keychain
- KeychainPassword: *****
- VerboseLog: false
Creating AppstoreConnectAPI client
the client created for https://api.appstoreconnect.apple.com/
Analyzing project
Failed to analyze project: failed to find the main target of the scheme (MTinfo 3000 TST): failed to find scheme (MTinfo 3000 TST) in project: scheme MTinfo 3000 TST not found in MTinfo 3000 TST
```

**Comment by rachid-debu** [comment]
hi @mac89 sorry for late answer. but I've managed to solve my issue by using the older version of this bitrise step : 
https://github.com/bitrise-steplib/steps-ios-auto-provision 
it is the basic step `iOS Auto Provision` (without app store connect API).
for this one to work, you will have to provide your apple credentials in your bitrise settings page.

let me know if this worked for you.

@bitce this new version of the step seem to have some issues, maybe needs to be investigated.

**Comment by mac89** [comment]
> hi @mac89 sorry for late answer. but I've managed to solve my issue by using the older version of this bitrise step :
> https://github.com/bitrise-steplib/steps-ios-auto-provision
> it is the basic step `iOS Auto Provision` (without app store connect API).
> for this one to work, you will have to provide your apple credentials in your bitrise settings page.
> 
> let me know if this worked for you.
> 
> @bitce this new version of the step seem to have some issues, maybe needs to be investigated.

I resorted to this as well. I perused through the source code, but could not find the cause.

**Comment by Tallyb** [comment]
This issue was closed without resolution to the step itself (the appconnect one). Can this be reopened until a proper solution exists?

**Comment by ceeK** [comment]
I've recently spent a day or so debugging this Bitrise step and this was one of my issues. The problem as far as I could see it is that this step doesn't load the workspace; it loads the project. Other tools produced by Bitrise [check if the path is a workspace or project and open the correct one](https://github.com/bitrise-io/xcode-project/blob/master/project.go#L18-L22), but this step doesn't. As all of our schemes are defined within the workspace and not within each project it fails to find the defined scheme. 

I got this step running locally using the Bitrise CLI and was playing around with it. Indeed changing the logic to open the workspace via (`xcworkspace.Open(pth)`) rather than the project (`xcodeproj.Open(pth)`) results in the scheme being visible. 

The workaround for us was to copy-paste the relevant scheme from `Foobar.xcworkspace/xcshareddata/xcschemes` to `Foobar.xcproj/xcshareddata/xcschemes`. This step will then find the scheme and everything works as expected. I did think about submitting a PR but my Go isn't fantastic unfortunately.

**Comment by coreymcmahon** [comment]
@ceeK We experienced the same issue. We needed to switch to `ios-auto-provision-appstoreconnect` because of Apple's changes to authentication in February 2021 so the "solution" above (reverting to the old build username-password version of the build step) was not possible.

I'd like to say thanks for the information you posted in the comment above as it allowed us to find a work-around which unblocked our build process. Hopefully @rachid-debu and/or the Bitrise team can fix this in the build step as it seems broken from our perspective (at least for any non-Capacitor Ionic projects).

For other developers who are currently blocked on this, we fixed it by adding another "Script" build step before "iOS Auto Provision with App Store Connect API":

![image](https://user-images.githubusercontent.com/700486/107932563-48214800-6fb0-11eb-8f5f-2caa45c78708.png)

In this script we simply copy all the schemes from `.xcworkspace` into `.xcodeproj`.

```
echo "Copying Workspace schemes to Project..."
mkdir -p platforms/ios/YOUR_SCHEME.xcodeproj/xcshareddata/xcschemes
cp -Rv platforms/ios/YOUR_SCHEME.xcworkspace/xcshareddata/xcschemes/* platforms/ios/YOUR_SCHEME.xcodeproj/xcshareddata/xcschemes/
echo "Done."
```

**Comment by rogerluan** [comment]
Hey all, thanks for everyone that contributed to the resolution of this issue. I believe someone at Bitrise should take a look and apply the fix that @ceeK already spent some hours investigating and debugging - that behavior doesn't seem to be "intentional" AFAICT :) I'd normally open a PR to fix this issue but since there's a decently clean workaround… 😁 

@coreymcmahon thanks for the very handy script! 💯 

@rachid-debu could you reopen this issue? 🙏

**Comment by godrei** [comment]
Hi all, we released a fix for this issue in version [1.0.2](https://github.com/bitrise-steplib/steps-ios-auto-provision-appstoreconnect/releases/tag/1.0.2). 

Please update to the latest version of the step, if you need more info/help just let us know by opening a new issue!
